module/applications/actor/type-config.mjs

```diff
@@ -15,31 +15,41 @@ export default class ActorTypeConfig extends FormApplication {
       choices: {},
       allowCustom: true,
       minimum: 0,
-      maximum: null
+      maximum: null,
+      sheetConfig: false
     });
   }
 
   /* -------------------------------------------- */
 
-  /** @inheritDoc */
+  /** @inheritdoc */
   get title() {
     return `${game.i18n.localize("DND5E.CreatureTypeTitle")}: ${this.object.name}`;
   }
 
   /* -------------------------------------------- */
 
-  /** @override */
+  /** @inheritdoc */
   get id() {
     return `actor-type-${this.object.id}`;
   }
 
   /* -------------------------------------------- */
 
-  /** @override */
-  getData(options={}) {
+  /**
+   * Key path to where the type is stored on the document.
+   * @type {string}
+   */
+  get keyPath() {
+    return this.document instanceof Actor ? "system.details.type" : "system.type";
```

Comment:
Instead of having to deduce this key path from within the context of an `ActorTypeConfig` instance, would be better to pass in the config as an `option` to the constructor when we create a `new ActorTypeConfig`. i.e.
```js
new ActorTypeConfig(race, {keyPath: "system.details.type"});
new ActorTypeConfig(actor, {keyPath: "system.type"});
```